website/shared/utils.js

```diff
@@ -50,4 +50,41 @@ const utilsModule = (function(window) {
   };
 })(window);
 
+
+//create a copy button for listingblock
+
+var numberOfSnippet = document.getElementsByClassName("listingblock").length;
+var listingblock;
+var divContent = [];
+var pre=[];
+
+for (let i =0; i<numberOfSnippet;i++){
+        
+        var button = document.createElement("button");
+        button.innerHTML = '<img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACcAAAAnCAYAAACMo1E1AAAAAXNSR0IArs4c6QAAAbdJREFUWEft2CuIVVEUBuBvfOCrCaJJ7FNmkgaxqcWigs1BTOIUi4/kIwkiaBDUJFoEccBiMJlkUIMIU4w2DSaRwQcoS/bAVu/d3rvPPWn2hlvO2utf//7PWuvsu6Z0X2twDPOYxg+8xU087QI/1cUZa3EPx4fg3MFp/KyJ05XcBVz9T+ATuD9JckF6L2awuQB8DluT/SseYSOOJFXD9B63Cxif8Tr9/tg2SLktWMDBMU97CneTz2VcGtP/IeZSzv52HUTuBs6MCRzbQ+kXye8oHldgnMe1Fb9B5OI17KwAjpOfxPqk/P4KjFfYXSKXV9YtXMyCvMO2QtBviNayrrBnCfsye1R0tKJYH7FjVHLXcTYD+oDtFYrkLtEDo9BW1oOsFTVyJXFXp3LxmdrUMec+pWqeeM515DXQfWIF0cjV9rmmXFNuxBxo1TqiUP9sa8o15fKb8N+XzVp1Sn4t52pVbco15Vq19vG/tTavVkefW04ztjhtDFae9yFXhnkojWvjUQyRdpVuwovY0zOhYfBPcLhELkZXz4bM7vrk/D2Nv96UyIXtAK5gFhv6ZIQvaeQa8+WXeaxfGhynKM2lswAAAAAASUVORK5CYII="/>';
+        button.classList.add("copy-code-button");
+        
+        button.setAttribute("id",i);
```

Comment:
Rather than just a numeric ID, it is a good practice to prefix them with some meaningful string. For example, you can set the id attribute as `"copy-code-button-" + i`  here